presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1172,21 +1173,27 @@ private static Domain buildColumnDomain(ColumnHandle column, List<HivePartition>
     @Override
     public Optional<ConnectorNewTableLayout> getInsertLayout(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
-        HivePartitionResult hivePartitionResult = partitionManager.getPartitions(metastore, tableHandle, Constraint.alwaysTrue());
-        if (!hivePartitionResult.getBucketHandle().isPresent()) {
+        HiveTableHandle hiveTableHandle = (HiveTableHandle) tableHandle;
```

Comment:
Here is the related code for fetching the table in [https://github.com/prestodb/presto/blob/2f148fd21a9431a67c038eee68807d67ed7829c6/presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java#L205](HivePartitionManager. getTable) (called by  `getPartitions`):

```
        Optional<Table> target = metastore.getTable(tableName.getSchemaName(), tableName.getTableName());
        if (!target.isPresent()) {
            throw new TableNotFoundException(tableName);
        }
        Table table = target.get();

        String protectMode = table.getParameters().get(ProtectMode.PARAMETER_NAME);
        if (protectMode != null && getProtectModeFromString(protectMode).offline) {
            throw new TableOfflineException(tableName, false, null);
        }

        String prestoOffline = table.getParameters().get(PRESTO_OFFLINE);
        if (!isNullOrEmpty(prestoOffline)) {
            throw new TableOfflineException(tableName, true, prestoOffline);
        }
```

Not sure if all the checks ( `protectMode` and `prestoOffline`) are required.